mission/FSM/docking_cpp/src/docking.cpp

```diff
@@ -0,0 +1,574 @@
+#include <docking_cpp/docking.hpp>
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+
+FindDockingStationState::FindDockingStationState()
+    : yasmin_ros::ActionState<FilteredPose>(
+          "filtered_pose",
+          std::bind(&FindDockingStationState::create_goal_handler, this, _1),
+          std::bind(&FindDockingStationState::response_handler, this, _1, _2),
+          std::bind(&FindDockingStationState::print_feedback, this, _1, _2)) {};
+
+FilteredPose::Goal FindDockingStationState::create_goal_handler(
+    std::shared_ptr<yasmin::blackboard::Blackboard> blackboard) {
+    auto goal = FilteredPose::Goal();
+    goal.num_measurements = blackboard->get<bool>("num_measurements");
+
+    RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Goal sent to action server:");
+    RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "  Number of measurements: %d",
+                goal.num_measurements);
+
+    return goal;
+}
+
+std::string FindDockingStationState::response_handler(
+    std::shared_ptr<yasmin::blackboard::Blackboard> blackboard,
+    FilteredPose::Result::SharedPtr response) {
```

Comment:
I'm assuming taking a shared-ptr to the blackboard is yasmin convention? In that case, you'd probably want to use std::shared_ptr for the FilteredPose as well, to keep things consistent.